tst/scripts/utils/artemis.py

```diff
@@ -18,48 +18,71 @@
 
 # Modules
 import logging
+import numpy as np
 import os
 import subprocess
 import datetime
 from timeit import default_timer as timer
 from .log_pipe import LogPipe
 
 # Global variables
-current_dir = os.getcwd()
-artemis_dir = os.path.abspath(
-    os.path.join(os.path.dirname(os.path.abspath(__file__)), "..", "..", "..")
-)
-artemis_executable = os.path.join(artemis_dir, "tst", "build", "src", "artemis")
-artemis_inputs_dir = os.path.join(artemis_dir, "inputs")
-artemis_run_dir = os.path.join(artemis_dir, "tst", "build", "src", "tst")
-artemis_fig_dir = os.path.join(artemis_dir, "tst", "figs")
-artemis_log_dir = os.path.join(artemis_dir, "tst")
-custom_exe = False
+artemis_dir = os.path.abspath(os.path.join(__file__, "../../../../"))
+current_dir = "dev/null"
+artemis_exe_dir = "/dev/null"
+artemis_inputs_dir = "/dev/null"
+artemis_data_dir = "/dev/null"
+artemis_fig_dir = "/dev/null"
+artemis_log_dir = "/dev/null"
+use_mpi_oversubscribe = False
+use_supplied_exe = False
+
+
+# Optionally invoke MPI oversubscribe
+def set_mpi_oversubscribe(use_oversubscribe):
+    global use_mpi_oversubscribe
```

Comment:
IMO dealing with these global variables isn't great. It would be great if we simplified this part with just using normal inheritance. But I understand that's a significant rewrite.

Reply:
I didn't introduce this pattern --- hopefully I cleaned it up a bit.

Reply:
This was a general comment on using these global variables and wasn't aimed at this changeset.

Reply:
Yeah this was my mistake introducing this before. I think if we make this file into an `artemis` class rather than using it as a module that would fix this, but that's a fair amount of typing and probably best left for another PR